### ISSUE TITLE
Stoke .o support

### ISSUE THREAD

**Issue by Ch0ronomato** [StanfordPL/stoke#925]
Does the stoke workflow support non-binary files? Like .o files? My particular use case is needing to optimize a .o

**Comment by stefanheule** [comment]
`stoke testcase`, `stoke extract` and `stoke replace` are the only main tools that work with binaries, and their job is to extract the function to be optimize (or put it back after) or get test data for it.  However, you don't need to get the information this way, you can also dissassmble the object file in a different way and give that to stoke.  Similarly with test-cases, if you can construct them in another way (e.g. using `stoke testcase --target /path/to/func` if you want random test data), then you can use `stoke optimize/synthesize` with that.

In short: There is no special support for .o files, but the input formats that most of STOKE expects can be derived from what you have.

**Comment by Ch0ronomato** [comment]
If we went this route, how could go about replacing the .o with the output from synthesize? Or does optimize handle this?

**Comment by stefanheule** [comment]
We don't currently have any tools to do that, but it shouldn't be too hard to write code to do this: Either use a linker to re-link everything (including the new code found by STOKE), or modify the `stoke replace` tool to work with .o files.  That shouldn't be too hard, the operation is very similar.

**Comment by Ch0ronomato** [comment]
Awesome, thanks so much Stefan!

I'll close this ticket.

**Comment by bchurchill** [comment]
It might actually already work.  We don't test for this, but I know
"stoke extract" will work out-of-the box on .o files.  "stoke replace"
might already work, but as Stefan says it could require tweaking. 
"stoke testcase" definitely won't work, since the very nature of the
tool requires an executable.
